File: tests/src/com.google.protobuf/protobuf-javalite/3.25.3/src/test/java/com_google_protobuf/protobuf_javalite/SchemaUtilTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_google_protobuf.protobuf_javalite;

import static org.assertj.core.api.Assertions.assertThat;

import com.google.protobuf.Struct;
import com.google.protobuf.Value;
import org.junit.jupiter.api.Test;

public class SchemaUtilTest {
    @Test
    void serializesAndParsesLiteMapMessage() throws Exception {
        Value value = Value.newBuilder().setStringValue("protobuf-javalite").build();
        Struct message = Struct.newBuilder().putFields("library", value).build();

        Struct parsed = Struct.parseFrom(message.toByteArray());

        assertThat(parsed.getFieldsOrThrow("library").getStringValue())
                .isEqualTo("protobuf-javalite");
    }
}

File: tests/src/org.apache.pekko/pekko-protobuf-v3_2.13/1.6.0/src/test/scala/org_apache_pekko/pekko_protobuf_v3_2_13/SchemaUtilTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_pekko.pekko_protobuf_v3_2_13

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class SchemaUtilTest {
  @Test
  def descriptorSchemaLooksUpMapDefaultEntryHolder(): Unit = {
    val message: SchemaUtilMapMessage = SchemaUtilMapMessage.getDefaultInstance

    message.parseEmptyInputThroughSchema()

    assertThat(message.getDescriptorForType.getFullName)
      .isEqualTo("schema_util.SchemaUtilMapMessage")
    assertThat(message.getLabelsCount).isZero()
    assertThat(message.getLabelsMap).isEmpty()
  }
}
